src/main/java/org/aion/wallet/dto/AccountDTO.java

```diff
@@ -79,6 +84,26 @@ public BufferedImage getQrCode() {
         return qrCode;
     }
 
+    public SortedSet<TransactionDTO> getTransactionsSnapshot() {
+        return new TreeSet<>(Collections.unmodifiableSortedSet(transactions));
```

Comment:
I would rather return a `Collections.unmodifiableSortedSet(new TreeSet(transactions));` Otherwise there's no point in using the unmodifiable wrapper and we can return the `new TreeSet(transactions)` directly.

Though it would probably make more sense to returnit as an unmodifiable collection (as you said in a different review)